public/app/features/dashboard/panel_editor/QueryOptions.tsx

```diff
@@ -70,8 +70,9 @@ export class QueryOptions extends PureComponent<Props, State> {
       name: 'maxDataPoints',
       tooltipInfo: (
         <>
-          The maximum data points the query should return. For graphs this is automatically set to one data point per
-          pixel.
+          The maximum data points the query should return. This is used to calculate a reasonable interval that is
```

Comment:
it's actually not true, I would love to figure out an away to support maxDataPoints for every data source (those that use interval), think maxDataPoints should override the interval
https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/state/PanelQueryRunner.ts#L172